src/web/js/check-ui.js

```diff
@@ -24,246 +24,426 @@
-                               ch:   get(get(checkBlock, "loc"), "end-column") },
-                             { inclusiveLeft:false,
-                               inclusiveRight:false,
-                               type:"range" });
-            header.on("click", function(e){
-              var handleLoc = handle.find();
-              if(source === "definitions://" && handleLoc !== undefined) {
-                editors[source].scrollIntoView(handleLoc.from.line, 100);
-              } else if (source.indexOf("interactions") != -1) {
-                editors[source].getWrapperElement().scrollIntoView(true);
-              }
+        return PassingTestSkeleton;
+      }();
+
+      var expandedCheckBlock = undefined;
+
+      var CheckBlockSkeleton = function () {
+        function CheckBlockSkeleton(name, loc, tests, error) {
+          var _this = this;
+
+          var container = document.createElement("div");
+          var testList = document.createElement("div");
+          var testFrag = document.createDocumentFragment();
+          var header = document.createElement("header");
+          var summary = document.createElement("span");
+
+          for (var i = 0; i < tests.skeletons.length; i++) {
+            var test = tests.skeletons[i];
+            test.block = this;
+            testFrag.appendChild(test.container);
+          }
+
+          if (error !== undefined) {
+            summary.textContent = "An unexpected error halted the check-block before Pyret was finished with it. " + "Some tests may not have run.";
+            var errorTestsSummary = document.createTextNode("Before the unexpected error, " + tests.executed + (tests.executed === 0 ? " tests " : " test ") + "in this block ran" + (tests.executed > 0 ? " (" + tests.passing + " passed):" : "."));
+            testList.appendChild(errorTestsSummary);
+          } else {
+            summary.textContent = tests.executed == 1 && tests.passing == 1 ? "The test in this block passed."
+            // Only one test in block; it fails
+            : tests.executed == 1 && tests.passing == 0 ? "The test in this block failed." : tests.executed == 0 ?
+            //  Huh, a block with no tests?
+            "There were no tests in this block!" : tests.executed == tests.passing ?
+            //  More than one test; all pass.
+            "All " + tests.executed + " tests in this block passed."
+            //  More than one test; some pass
+            : tests.passing + " out of " + tests.executed + " tests passed in this block.";
+          }
+
+          testList.classList.add("check-block-tests");
+          summary.classList.add("check-block-summary");
+
+          header.classList.add("check-block-header");
+          header.title = "Click to view test results.";
+          header.appendChild(makeNameHandle(name, loc, error !== undefined ? "hsl(0, 100%, 85%)" : tests.executed == tests.passing ? "hsl(88, 50%, 76%)" : "hsl(45, 100%, 85%)").anchor);
```

Comment:
A lot of these quirks address code that was introduced pre-pull request but called out in the diffs because so much was re-intended, moved around, etc. Since this pull request comes after the merge to master, a goal was to change as little as possible and to limit the changes I did introduce to things that overall simplified the code and reduced the surface-area for bugs.